migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java

```diff
@@ -0,0 +1,116 @@
+package com.redhat.lightblue.migrator;
+
+import java.util.Map;
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Works like a breakpoint in a debugger. If stop is called, execution
+ * stops when run() is called until resume() is called by another
+ * thread.
+ */
+public class Breakpoint {
```

Comment:
It is used in unit tests, to stop the execution of a thread at a certain
point, inspect thread state, and then resume exection.

On Wed, Jun 3, 2015 at 7:04 AM, Dennis Crissman <notifications@github.com>
wrote:

> In migrator/src/main/java/com/redhat/lightblue/migrator/Breakpoint.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/197#discussion_r31619697>
> :
>
> > @@ -0,0 +1,116 @@
> > +package com.redhat.lightblue.migrator;
> > +
> > +import java.util.Map;
> > +import java.util.HashMap;
> > +
> > +import org.slf4j.Logger;
> > +import org.slf4j.LoggerFactory;
> > +
> > +/**
> > + * Works like a breakpoint in a debugger. If stop is called, execution
> > + * stops when run() is called until resume() is called by another
> > + * thread.
> > + */
> > +public class Breakpoint {
>
> Is the intention of this class for debugging purposes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/197/files#r31619697>
> .
>